### PR TITLE
[VitisAI] Add processing for sessionOptions.AppendExecutionProvider("VitisAI", options)

### PR DESCRIPTION
**Pull request by zz002**
### Description

[VitisAI] Add processing for sessionOptions.AppendExecutionProvider("VitisAI", options)

### CODE REVIEW

File: onnxruntime/core/session/provider_registration.cc [hunk 1]
```diff
@@ -148,6 +148,8 @@
 #else
     status = create_not_supported_status();
 #endif
+  } else if (strcmp(provider_name, "VitisAI") == 0) {
+    OrtApis::SessionOptionsAppendExecutionProvider_VitisAI(options, provider_options_keys, provider_options_values, num_keys);
```
[the remark this answers is not in the document]
Comment (jywu-msft) [reply onnxruntime/core/session/provider_registration.cc:152]: this doesn't seem to follow same pattern as other EP's. is there a factory creator function for VitisAI EP?

[the remark this answers is not in the document]
Comment (zz002) [reply onnxruntime/core/session/provider_registration.cc]: https://github.com/microsoft/onnxruntime/pull/21839/files#diff-ab256eeda40b929f50561f447d57d800b92e4b5f1977b1d0487da60349c190d1R498
What is called here is the interface in the link above

**Review by tianleiwu — APPROVED** [review]

**Review by jywu-msft — APPROVED** [review]

**Comment by jywu-msft** [comment]
can you merge with latest main branch?

**Comment by zz002** [comment]
> can you merge latest main to address the Linux OpenVINO CI Pipeline failure?

done

**Pull request merged**